src/cc/table_desc.h

```diff
@@ -93,10 +81,10 @@ class TableDesc {
         leaf_snprintf(nullptr),
         is_shared(false),
         is_extern(false) {}
-  TableDesc(const std::string &name, FileDesc &&fd, int type, size_t key_size,
+  TableDesc(const std::string &name, int fd, int type, size_t key_size,
```

Comment:
@drzaeus77 Sorry I'm a bit confused now. My main goal is to mark the int constructor as `explict` to avoid this kind of implicit conversions. It doesn't seem that we actually want to duplicate the fd here, so what have I missed? I construct the temporary object in the call site with FD, and move it to the `TableDesc` local member. That seems to be exactly the behavior as your current implementation.

I also can't reproduce the test failure locally. It seems to be only happening in fedora hosts? Is that compiler related?

Sorry for all the trouble and thanks a lot for your patience!